File: lab19/task3.cpp
#include <iostream>
using namespace std;

int factorial(int n) {
    if (n == 0) {
        return 1;
    } else {
        return n * factorial(n - 1);
    }
}

int main() {
    // Example test cases
    cout << "Factorial = " << factorial(5) << endl;  // Output: Factorial = 120
    cout << "Factorial = " << factorial(0) << endl;  // Output: Factorial = 1
    
    return 0;
}

File: lab19/task3.py
def factorial(n):
    if n == 0:
        return 1
    else:
        return n * factorial(n - 1)

# Example test cases
if __name__ == "__main__":
    print("Factorial =", factorial(5))  # Output: Factorial = 120
    print("Factorial =", factorial(0))  # Output: Factorial = 1
